File: .threads/pinger.py
from threading import Thread
from time import sleep

class Pinger(Thread):
    def __init__(self):
        super(Pinger, self).__init__()
        self.daemon = True
        self.__killed = False

    def run(self):
        i = 1
        while not self.__killed:
            print(f"{i}. ping")
            sleep(1)
            i += 1

    def kill(self):
        self.__killed = True
        print("pinger killed")

File: .threads/Pinger.java
import java.lang.Runnable;
import java.lang.Exception;

public class Pinger extends Thread {
    private boolean killed = false;

    public Pinger() {
    }

    @Override
    public void run() {
        try {
            int i = 1;
            while(!killed) {
                System.out.println(i + ". ping");
                Thread.sleep(1000);
                i++;
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void kill() {
        killed = true;
        System.out.println("Pinger killed");
    }
}

